### PR TITLE
Live tests for Communication CallingServer SDK.

### PR DESCRIPTION
**Pull request by moirf**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs [hunk 1]
```diff
@@ -101,6 +107,359 @@ public async Task RunCreateAddRemoveHangupScenarioTests()
...
+                await WaitForOperationCompletion().ConfigureAwait(false);
+                AddParticipantResult addParticipantResult = await AddParticipantOperation(callConnection, userId).ConfigureAwait(false);
+                Assert.NotNull(addParticipantResult);
+
+                // Mute Participant
+                await WaitForOperationCompletion().ConfigureAwait(false);
+                await MuteParticipantOperation(callConnection, userId).ConfigureAwait(false);
+
+                // Get Participant
+                await WaitForOperationCompletion().ConfigureAwait(false);
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:140]: mute is synchronous, so no need to wait

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs [hunk 2]
```diff
@@ -101,6 +107,359 @@ public async Task RunCreateAddRemoveHangupScenarioTests()
...
+                var getMutedParticipant = await GetParticipantOperation(callConnection, userId).ConfigureAwait(false);
+                Assert.IsTrue(getMutedParticipant.IsMuted == true);
+
+                // Unmute Participant
+                await WaitForOperationCompletion().ConfigureAwait(false);
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:145]: no need to wait here

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs [hunk 3]
```diff
@@ -101,6 +107,359 @@ public async Task RunCreateAddRemoveHangupScenarioTests()
...
+                // Add Participant
+                await WaitForOperationCompletion().ConfigureAwait(false);
+                AddParticipantResult addParticipantResult = await AddParticipantOperation(callConnection, userId).ConfigureAwait(false);
+                Assert.NotNull(addParticipantResult);
+
+                // Mute Participant
+                await WaitForOperationCompletion().ConfigureAwait(false);
+                await MuteParticipantOperation(callConnection, userId).ConfigureAwait(false);
+
+                // Get Participant
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:139]: update comment

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs [hunk 4]
```diff
@@ -101,6 +107,359 @@ public async Task RunCreateAddRemoveHangupScenarioTests()
...
+                await WaitForOperationCompletion().ConfigureAwait(false);
+                var getMutedParticipant = await GetParticipantOperation(callConnection, userId).ConfigureAwait(false);
+                Assert.IsTrue(getMutedParticipant.IsMuted == true);
+
+                // Unmute Participant
+                await WaitForOperationCompletion().ConfigureAwait(false);
+                await UnmuteParticipantOperation(callConnection, userId).ConfigureAwait(false);
+
+                // Get Participant
+                await WaitForOperationCompletion().ConfigureAwait(false);
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:149]: no need to wait here

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs [hunk 5]
```diff
@@ -101,6 +107,359 @@ public async Task RunCreateAddRemoveHangupScenarioTests()
...
+                var getUnmutedParticipant = await GetParticipantOperation(callConnection, userId).ConfigureAwait(false);
+                Assert.IsTrue(getUnmutedParticipant.IsMuted == false);
+
+                // Get Participants
+                await WaitForOperationCompletion().ConfigureAwait(false);
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:154]: no need to wait here

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs [hunk 6]
```diff
@@ -101,6 +107,359 @@ public async Task RunCreateAddRemoveHangupScenarioTests()
...
+                var getParticipants = await GetParticipantsOperation(callConnection).ConfigureAwait(false);
+                Assert.IsTrue(getParticipants.Count() > 2);
+
+                // Remove Participant
+                await WaitForOperationCompletion().ConfigureAwait(false);
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:159]: no need to wait here

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs [hunk 7]
```diff
@@ -101,6 +107,359 @@ public async Task RunCreateAddRemoveHangupScenarioTests()
...
+            try
+            {
+                string userId = GetFixedUserId(TestEnvironment.UserIdentifier);
+
+                // Add Participant
+                await WaitForOperationCompletion().ConfigureAwait(false);
+                AddParticipantResult addParticipantResult = await AddParticipantOperation(callConnection, userId).ConfigureAwait(false);
+                Assert.NotNull(addParticipantResult);
+
+                // Get Call
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:200]: this comment is for the next line of getcall, and not for the addparticipant operation wait

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs [hunk 8]
```diff
@@ -101,6 +107,359 @@ public async Task RunCreateAddRemoveHangupScenarioTests()
...
+                await WaitForOperationCompletion().ConfigureAwait(false);
+                var getCallConnection = await GetCallOperation(callConnection).ConfigureAwait(false);
+                Assert.AreEqual(getCallConnection.CallConnectionId, callConnection.Value.CallConnectionId);
+
+                // Hold Participant
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:205]: here and at other places, pl make sure the comment reflects the line beside it or right under it.

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs [hunk 9]
```diff
@@ -101,6 +107,359 @@ public async Task RunCreateAddRemoveHangupScenarioTests()
...
+                await WaitForOperationCompletion().ConfigureAwait(false);
+                await HoldParticipantOperation(callConnection, userId).ConfigureAwait(false);
+
+                // Resume Participant
+                await WaitForOperationCompletion().ConfigureAwait(false);
+                await ResumeParticipantOperation(callConnection, userId).ConfigureAwait(false);
+
+                // Remove Participant
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:213]: for none of these operations, there is no need to wait

Comment (ritanshu07) [reply sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:213]: Made changes

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs (hunk 8 above)
Comment (ritanshu07) [reply sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:205]: Made changes

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs (hunk 7 above)
Comment (ritanshu07) [reply sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs]: Made changes

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs (hunk 6 above)
Comment (ritanshu07) [reply sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs]: Made changes

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs (hunk 5 above)
Comment (ritanshu07) [reply sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs]: Made changes

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs (hunk 4 above)
Comment (ritanshu07) [reply sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs]: Made changes

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs (hunk 2 above)
Comment (ritanshu07) [reply sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs]: Made changes

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs (hunk 1 above)
Comment (ritanshu07) [reply sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs]: Made changes

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs [hunk 10]
```diff
@@ -18,20 +18,45 @@ namespace Azure.Communication.CallingServer.Tests
...
+            if (Mode != RecordedTestMode.Playback)
+            {
+                return TestEnvironment.AudioFileUri;
+            }
+            return TestEnvironment.AudioFileUrl;
+        }
+
+        protected string GetDownloadEndPointUrl()
+        {
+            if (Mode != RecordedTestMode.Playback)
```
Comment (pakrym) [inline sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs:50]: You can pass your placeholder value to .IsSecret() call in the TestEnvironment

```
        public string DownloadEndPoint => GetRecordedVariable(DownloadUrl, options => options.IsSecret(DOWNLOAD_END_POINT));
```

That way you don't have to have your own logic based on the mode.

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerTestEnvironment.cs [hunk 11]
```diff
@@ -16,6 +17,50 @@ public class CallingServerTestEnvironment : CommunicationTestEnvironment
...
+        public const string AudioUrl = "AUDIO_FILE_URL";
+
+        public const string DeleteUrl = "DELETE_ENDPOINT";
+
+        public const string AsyncDeleteUrl = "ASYNC_DELETE_ENDPOINT";
+
+        /// <summary>
+        /// The delete url required to delete the file.
+        /// </summary>
+        public string DeleteEndPoint => GetRecordedVariable(DeleteUrl, options => options.IsSecret());
```
Comment (pakrym) [inline sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerTestEnvironment.cs:37]: Are all of these values really secret?

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs [hunk 12]
```diff
@@ -62,6 +67,7 @@ public async Task RunCreatePlayCancelHangupScenarioTests()
             finally
             {
                 // Hang up the Call, there is one call leg in this test case, hangup the call will also delete the call as the result.
+                await WaitForOperationCompletion().ConfigureAwait(false);
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:70]: we dont need to wait for operation before hanging up. please remove this from other tests as well

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs [hunk 13]
```diff
@@ -79,12 +85,366 @@ public async Task RunCreateAddRemoveHangupScenarioTests()
...
+
+            CallingServerClient client = CreateInstrumentedCallingServerClientWithConnectionString();
+
+            // Establish a call
+            var callConnection = await CreateCallConnectionOperation(client).ConfigureAwait(false);
+
+            try
+            {
+                string userId = GetFixedUserId(TestEnvironment.UserIdentifier);
+                await WaitForOperationCompletion().ConfigureAwait(false);
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:132]: move this to the line above it, as we are waiting for the callconnection operation. Also pl move the call creation to inside the try block for this and other tests

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs [hunk 14]
```diff
@@ -79,12 +85,366 @@ public async Task RunCreateAddRemoveHangupScenarioTests()
...
+            // Establish a call
+            var callConnection = await CreateCallConnectionOperation(client).ConfigureAwait(false);
+
+            try
+            {
+                string targetParticipant = GetFixedUserId(TestEnvironment.UserIdentifier);
+                await WaitForOperationCompletion().ConfigureAwait(false);
+
+                // Transfer Call
+                await TransferCallToParticipantOperation(callConnection, targetParticipant).ConfigureAwait(false);
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:286]: we need to wait for transfer operation

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs [hunk 15]
```diff
@@ -79,12 +85,366 @@ public async Task RunCreateAddRemoveHangupScenarioTests()
...
+            var callConnection = await CreateCallConnectionOperation(client).ConfigureAwait(false);
+
+            await WaitForOperationCompletion().ConfigureAwait(false);
+
+            try
+            {
+                // Keep Alive
+                await KeepAliveOperation(callConnection).ConfigureAwait(false);
+
+                await WaitForOperationCompletion().ConfigureAwait(false);
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:252]: waiting not needed for keepalive

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs [hunk 16]
```diff
@@ -79,12 +85,366 @@ public async Task RunCreateAddRemoveHangupScenarioTests()
...
+
+                // Add Participant
+                AddParticipantResult addParticipantResult = await AddParticipantOperation(callConnection, userId).ConfigureAwait(false);
+                Assert.NotNull(addParticipantResult);
+
+                // Play Audio To Participant
+                var playAudioResult = await PlayAudioToParticipantOperation(callConnection, userId).ConfigureAwait(false);
+
+                await WaitForOperationCompletion().ConfigureAwait(false);
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:355]: we need separate waits for add participant and playaudiotoparticipant

File: sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs [hunk 17]
```diff
@@ -79,12 +85,366 @@ public async Task RunCreateAddRemoveHangupScenarioTests()
...
+
+                // Get Audio Routing Group
+                var getAudioRoutingResult = await GetAudioRoutingGroupOperation(callConnection, audioRoutingGroupId).ConfigureAwait(false);
+                Assert.IsTrue(getAudioRoutingResult.AudioRoutingMode == AudioRoutingMode.Multicast);
+                foreach (var target in getAudioRoutingResult.Targets)
+                {
+                    Assert.IsTrue(target.ToString() == userId);
+                }
+
+                await WaitForOperationCompletion().ConfigureAwait(false);
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallConnection/CallConnectionLiveTests.cs:422]: no need to wait for audioroutinggroup operations

File: sdk/communication/Azure.Communication.CallingServer/tests/CallingServerClient/CallingServerClientLiveTests.cs [hunk 18]
```diff
@@ -65,7 +67,7 @@ public async Task RunAllRecordingFunctionsScenarioTests(AuthMethod authMethod)
                 Assert.ThrowsAsync<RequestFailedException>(async () => await callingServerClient.GetRecordingStateAsync(recordingId).ConfigureAwait(false));
 
                 // Hang up the Call, there is one call leg in this test case, hangup the call will also delete the call as the result.
-                await SleepIfNotInPlaybackModeAsync().ConfigureAwait(false);
+                await WaitForOperationCompletion().ConfigureAwait(false);
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallingServerClient/CallingServerClientLiveTests.cs:70]: we dont have to wait for getting call recording state. pl fix in other places in code as well

File: sdk/communication/Azure.Communication.CallingServer/tests/CallingServerClient/CallingServerClientLiveTests.cs [hunk 19]
```diff
@@ -91,13 +93,14 @@ public async Task RunCreatePlayCancelHangupScenarioTests()
             try
             {
                 var callLocator = new GroupCallLocator(groupId);
+                await WaitForOperationCompletion().ConfigureAwait(false);
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallingServerClient/CallingServerClientLiveTests.cs:96]: no need to wait after creating calllocators. pl fix at other places in code as well

File: sdk/communication/Azure.Communication.CallingServer/tests/CallingServerClient/CallingServerClientLiveTests.cs [hunk 20]
```diff
@@ -112,7 +115,7 @@ public async Task RunCreatePlayCancelHangupScenarioTests()
             finally
             {
                 // Hang up the Call, there is one call leg in this test case, hangup the call will also delete the call as the result.
-                await SleepIfNotInPlaybackModeAsync().ConfigureAwait(false);
+                await WaitForOperationCompletion().ConfigureAwait(false);
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallingServerClient/CallingServerClientLiveTests.cs:118]: no need to wait in finally blocks, pl remove everywhere

File: sdk/communication/Azure.Communication.CallingServer/tests/CallingServerClient/CallingServerClientLiveTests.cs [hunk 21]
```diff
@@ -132,15 +135,276 @@ public async Task RunCreateAddRemoveHangupScenarioTests()
 
             try
             {
-                string userId = GetFixedUserId("0000000d-5e6a-3252-a808-4548220003b8");
+                string userId = GetFixedUserId(TestEnvironment.UserIdentifier);
+                await WaitForOperationCompletion().ConfigureAwait(false);
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallingServerClient/CallingServerClientLiveTests.cs:139]: what is this wait for? pl remove here and elsewhere in similar places

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs [hunk 22]
```diff
@@ -362,6 +416,323 @@ internal async Task RemoveParticipantOperation(CallingServerClient callingServer
             Assert.AreEqual(202, response.Status);
         }
         #endregion Snippet:Azure_Communication_ServerCalling_Tests_RemoveParticipantOperation
+
+        #region Snippet:Azure_Communication_ServerCalling_Tests_MuteParticipantOperation
+        internal async Task MuteParticipantOperation(CallConnection callConnection, string participantUserId)
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs:421]: mute doesnt create an operation. can we just call it MuteParticipant?

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs [hunk 23]
```diff
@@ -362,6 +416,323 @@ internal async Task RemoveParticipantOperation(CallingServerClient callingServer
...
+            Console.WriteLine("Performing mute participant operation to mute a participant");
+
+            var response = await callConnection.MuteParticipantAsync(new CommunicationUserIdentifier(participantUserId)).ConfigureAwait(false);
+
+            Assert.AreEqual(200, response.Status);
+        }
+        #endregion Snippet:Azure_Communication_ServerCalling_Tests_MuteParticipantOperation
+
+        #region Snippet:Azure_Communication_ServerCalling_Tests_UnmuteParticipantOperation
+        internal async Task UnmuteParticipantOperation(CallConnection callConnection, string participantUserId)
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs:432]: unmute doesnt create an operation. can we just call it UnmuteParticipant?

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs [hunk 24]
```diff
@@ -362,6 +416,323 @@ internal async Task RemoveParticipantOperation(CallingServerClient callingServer
...
+            Console.WriteLine("Performing unmute participant operation to unmute a participant");
+
+            var response = await callConnection.UnmuteParticipantAsync(new CommunicationUserIdentifier(participantUserId)).ConfigureAwait(false);
+
+            Assert.AreEqual(200, response.Status);
+        }
+        #endregion Snippet:Azure_Communication_ServerCalling_Tests_UnmuteParticipantOperation
+
+        #region Snippet:Azure_Communication_ServerCalling_Tests_GetParticipantOperation
+        internal async Task<CallParticipant> GetParticipantOperation(CallConnection callConnection, string participantUserId)
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs:443]: this doesnt create an operation. can we just call it GetParticipant?

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs [hunk 25]
```diff
@@ -362,6 +416,323 @@ internal async Task RemoveParticipantOperation(CallingServerClient callingServer
...
+            Console.WriteLine("Performing get participant operation to get a participant");
+
+            var response = await callConnection.GetParticipantAsync(new CommunicationUserIdentifier(participantUserId)).ConfigureAwait(false);
+
+            Assert.IsFalse(string.IsNullOrWhiteSpace(response.Value.ToString()));
+
+            return response.Value;
+        }
+
+        internal async Task<CallParticipant> GetParticipantOperation(CallingServerClient callingServerClient, CallLocator callLocator, string participantUserId)
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs:454]: this doesnt create an operation. can we just call it GetParticipant?

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs [hunk 26]
```diff
@@ -362,6 +416,323 @@ internal async Task RemoveParticipantOperation(CallingServerClient callingServer
...
+            var response = await callingServerClient.GetParticipantAsync(callLocator, new CommunicationUserIdentifier(participantUserId)).ConfigureAwait(false);
+
+            Assert.IsFalse(string.IsNullOrWhiteSpace(response.Value.ToString()));
+
+            return response.Value;
+        }
+        #endregion Snippet:Azure_Communication_ServerCalling_Tests_GetParticipantOperation
+
+        #region Snippet:Azure_Communication_ServerCalling_Tests_GetParticipantsOperation
+        internal async Task<IEnumerable<CallParticipant>> GetParticipantsOperation(CallConnection callConnection)
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs:467]: this doesnt create an operation. can we just call it GetParticipants?

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs [hunk 27]
```diff
@@ -362,6 +416,323 @@ internal async Task RemoveParticipantOperation(CallingServerClient callingServer
...
+            Console.WriteLine("Performing get participants operation to get participants");
+
+            var response = await callConnection.GetParticipantsAsync().ConfigureAwait(false);
+
+            Assert.IsFalse(string.IsNullOrWhiteSpace(response.Value.ToString()));
+
+            return response.Value;
+        }
+
+        internal async Task<IEnumerable<CallParticipant>> GetParticipantsOperation(CallingServerClient callingServerClient, CallLocator callLocator)
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs:478]: this doesnt create an operation. can we just call it GetParticipants?

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs [hunk 28]
```diff
@@ -362,6 +416,323 @@ internal async Task RemoveParticipantOperation(CallingServerClient callingServer
...
+            var response = await callingServerClient.GetParticipantsAsync(callLocator).ConfigureAwait(false);
+
+            Assert.IsFalse(string.IsNullOrWhiteSpace(response.Value.ToString()));
+
+            return response.Value;
+        }
+        #endregion Snippet:Azure_Communication_ServerCalling_Tests_GetParticipantsOperation
+
+        #region Snippet:Azure_Communication_ServerCalling_Tests_GetCallOperation
+        internal async Task<CallConnectionProperties> GetCallOperation(CallConnection callConnection)
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs:491]: this doesnt create an operation. can we just call it GetCall?

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs [hunk 29]
```diff
@@ -362,6 +416,323 @@ internal async Task RemoveParticipantOperation(CallingServerClient callingServer
...
+            var response = await callConnection.GetCallAsync().ConfigureAwait(false);
+
+            Assert.IsFalse(string.IsNullOrWhiteSpace(response.Value.ToString()));
+
+            return response.Value;
+        }
+        #endregion Snippet:Azure_Communication_ServerCalling_Tests_GetCallOperation
+
+        #region Snippet:Azure_Communication_ServerCalling_Tests_HoldParticipantOperation
+        internal async Task HoldParticipantOperation(CallConnection callConnection, string participantUserId)
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs:504]: this doesnt create an operation. can we just call it HoldParticipant?

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs [hunk 30]
```diff
@@ -362,6 +416,323 @@ internal async Task RemoveParticipantOperation(CallingServerClient callingServer
...
+            Console.WriteLine("Performing hold participant operation to hold a participant");
+
+            var response = await callConnection.HoldParticipantMeetingAudioAsync(new CommunicationUserIdentifier(participantUserId)).ConfigureAwait(false);
+
+            Assert.AreEqual(200, response.Status);
+        }
+        #endregion Snippet:Azure_Communication_ServerCalling_Tests_HoldParticipantOperation
+
+        #region Snippet:Azure_Communication_ServerCalling_Tests_ResumeParticipantOperation
+        internal async Task ResumeParticipantOperation(CallConnection callConnection, string participantUserId)
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs:515]: this doesnt create an operation. can we just call it ResumeParticipant?

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs [hunk 31]
```diff
@@ -362,6 +416,323 @@ internal async Task RemoveParticipantOperation(CallingServerClient callingServer
...
+            Console.WriteLine("Performing resume participant operation to resume a participant");
+
+            var response = await callConnection.ResumeParticipantMeetingAudioAsync(new CommunicationUserIdentifier(participantUserId)).ConfigureAwait(false);
+
+            Assert.AreEqual(200, response.Status);
+        }
+        #endregion Snippet:Azure_Communication_ServerCalling_Tests_ResumeParticipantOperation
+
+        #region Snippet:Azure_Communication_ServerCalling_Tests_KeepAliveOperation
+        internal async Task KeepAliveOperation(CallConnection callConnection)
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs:526]: this doesnt create an operation. can we just call it KeepAlive?

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs [hunk 32]
```diff
@@ -362,6 +416,323 @@ internal async Task RemoveParticipantOperation(CallingServerClient callingServer
...
+            Console.WriteLine("Performing keep alive operation");
+
+            var response = await callConnection.KeepAliveAsync().ConfigureAwait(false);
+
+            Assert.AreEqual(200, response.Status);
+        }
+        #endregion Snippet:Azure_Communication_ServerCalling_Tests_KeepAliveOperation
+
+        #region Snippet:Azure_Communication_ServerCalling_Tests_DeleteCallOperation
+        internal async Task DeleteCallOperation(CallConnection callConnection)
```
Comment (satyakonmsft) [inline sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs:537]: this doesnt create an operation. can we just call it DeleteCall?

**Review by satyakonmsft — APPROVED** [review]

**Pull request merged**